modules/node_groups/node_groups.tf

```
@@ -0,0 +1,49 @@
+resource "aws_eks_node_group" "workers" {
+  for_each = random_pet.node_groups
+
+  node_group_name = join("-", [var.cluster_name, each.key, each.value.id])
+
+  cluster_name  = var.cluster_name
+  node_role_arn = each.value.keepers.node_role_arn
+  subnet_ids    = split("|", each.value.keepers.subnet_ids)
+
+  scaling_config {
+    desired_size = lookup(local.node_groups_expanded[each.key], "desired_capacity", var.workers_group_defaults["asg_desired_capacity"])
+    max_size     = lookup(local.node_groups_expanded[each.key], "max_capacity", var.workers_group_defaults["asg_max_size"])
+    min_size     = lookup(local.node_groups_expanded[each.key], "min_capacity", var.workers_group_defaults["asg_min_size"])
+  }
+
+  ami_type        = lookup(each.value.keepers, "ami_type", null)
+  disk_size       = lookup(each.value.keepers, "disk_size", null)
+  instance_types  = [each.value.keepers.instance_type]
+  release_version = lookup(local.node_groups_expanded[each.key], "ami_release_version", null)
+
+  dynamic "remote_access" {
+    for_each = each.value.keepers.ec2_ssh_key != "" ? [{
+      ec2_ssh_key               = each.value.keepers.ec2_ssh_key
+      source_security_group_ids = compact(split("|", each.value.keepers.source_security_group_ids))
+    }] : []
+
+    content {
+      ec2_ssh_key               = remote_access.value["ec2_ssh_key"]
+      source_security_group_ids = remote_access.value["source_security_group_ids"]
+    }
+  }
+
+  version = lookup(local.node_groups_expanded[each.key], "version", null)
+
+  labels = merge(
```

I believe that you can replace this entire block with something like the following:
```suggestion
  labels = lookup(local.node_groups_expanded, "k8s_labels", null)
```

You can combine lines 42 and 43 below by doing something similar.

---

Those are functionally two different things as Terraform does not do deep merging of lists:
```
> merge({a = {a = "a"}}, {a = {aa = "aa"}})
{
  "a" = {
    "aa" = "aa"
  }
}
```

I thought the workers merged the default tags with the per workers tags but on double checking the code it's a simple fallback: [here](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/c17d17e2c251ce9f57fcab6a6822ca138f01f035/workers.tf#L132..L135). Although that's harder as they are lists of maps and merging there isn't easy.

---

Right.